allennlp/models/language_model.py

```diff
@@ -74,6 +75,23 @@ class LanguageModel(Model):
     contextualizer: ``Seq2SeqEncoder``
         Used to "contextualize" the embeddings. As described above,
         this encoder must not cheat by peeking ahead.
+
+        .. deprecated:: 0.8.2
```

Comment:
Question: To me this seems like we are ripping out quite a good api for one which is more complicated. We already have a trained `BidirectionalLanguageModel` which uses a bidirectional transformer, which is very useful to many people (e.g github issues/ Swabha is using it in her research), which makes me unsure that depreciating a key component of it is a good idea. Do we know that bidirectional language modeling of contiguous text is worth it? I can't imagine a downstream task where you actually need unlimited context (and even if there was one, how that would practically work). Is it worth ripping this into a separate repo, making the changes and double checking that contiguous text does something useful? 

Additionally, we now have a `BidirectionalLanguageModel` subclass which is subsumed by this code (I think). This is "bad code smell" to me when we have a subclass which only differs in constructor arguments and we only have a single example of how those arguments might differ (it triggers my `MultiClassMultipleChoiceMemoryNetwork(MemoryNetwork, MultiClass, MultipleChoice)` sense from [DeepQa](https://github.com/allenai/deep_qa/pull/389/files#diff-9b4839b38293d3faa5e7663cc69e7e3aL10)).

Reply:
Leaving the API decision to others, here's some motivation for enabling modeling of longer contexts: https://ai.googleblog.com/2019/01/transformer-xl-unleashing-potential-of.html.  They basically make a stateful transformer, showing pretty big gains.